coalib/files/Fileproxy.py

```diff
@@ -0,0 +1,89 @@
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq("filename")
+class Fileproxy:
+    """
+    The file proxy is a wrapper object to wrap every file. The file proxy is an
+    object that contains properties about the file currently processed (such as
+    the filename). It contains:
+
+        * The filename, this is the starting point for every other attribute.
+        * The content of the file as a string.
+        * The content of the file, line-splitted.
+
+    The equality of the object is checked just based on its filename.
+
+    To initialize a Fileproxy object for a file:
+
+    >>> fp = Fileproxy("./tests/files/FileproxyTestFiles/testfile.txt")
+
+    Now we can get the filename:
+
+    >>> fp.filename
+    './tests/files/FileproxyTestFiles/testfile.txt'
+
+    The number of lines in the file:
+
+    >>> len(fp)
+    1
+
+    Iterate through the lines of the file:
+
+    >>> for line in fp:
+    ...     print(line)
+    Old line
+    <BLANKLINE>
+
+    Get contents of the file as a string:
+
+    >>> fp.content
+    'Old line\\n'
+
+    Get all lines of the file as a list:
+    >>> fp.lines
+    ['Old line\\n']
+    """
+
+    def __init__(self, filename):
+        """
+        Constructs a new fileproxy object.
+
+        :param filename: The name of the file to load.
+        """
+        self.filename = filename
+        with open(self.filename, encoding="utf-8") as filehandle:
+            self.__lines = tuple(filehandle.readlines())
```

Comment:
have you done any performance testing on converting the returned list to a tuple?
As I have said previously, trying to prevent damage is not a good design.  If you want to write a proxy, do it in Cython.  Then you can really control access, and prevent damage.
A single underscore is sufficient.  PyLint and other tools warn about access to private attributes.  We should be using linters, rather than creating complexity in the code.